src/QsFmt/App/Program.fs

```diff
@@ -33,93 +34,107 @@ let run arguments inputs =
                     let newInputs =
                         let topLevelFiles = Directory.EnumerateFiles(input, "*.qs") |> List.ofSeq
 
-                        if arguments.RecurseFlag then
+                        if commandWithOptions.RecurseFlag then
                             topLevelFiles @ (Directory.EnumerateDirectories input |> List.ofSeq)
                         else
                             topLevelFiles
 
-                    newInputs |> doMany arguments
+                    newInputs |> processManyInputs
                 else
                     let source =
                         if input = "-" then
                             stdin.ReadToEnd()
                         else
-                            if arguments.BackupFlag then File.Copy(input, (input + "~"), true)
+                            if commandWithOptions.BackupFlag then File.Copy(input, (input + "~"), true)
                             File.ReadAllText input
 
                     let command =
-                        match arguments.CommandKind with
-                        | Update -> Formatter.update input arguments.QSharp_Version
-                        | Format -> Formatter.format arguments.QSharp_Version
-                        | UpdateAndFormat -> Formatter.updateAndFormat input arguments.QSharp_Version
+                        match commandWithOptions.CommandKind with
+                        | Update -> Formatter.update input commandWithOptions.QSharpVersion
+                        | Format -> Formatter.format commandWithOptions.QSharpVersion
+                        | UpdateAndFormat -> Formatter.updateAndFormat input commandWithOptions.QSharpVersion
 
                     match command source with
                     | Ok result ->
                         if input = "-" then printf "%s" result else File.WriteAllText(input, result)
-                        0
+                        { RunResult.Default with FilesProcessed = 1 }
                     | Error errors ->
                         // Change the "-" input to say "<Standard Input>" in the error
                         let input = if input = "-" then "<Standard Input>" else input
                         errors |> List.iter (eprintfn "%s, %O" input)
-                        1
+                        { RunResult.Default with ExitCode = ExitCode.SyntaxErrors; SyntaxErrors = errors }
             with
             | :? IOException as ex ->
                 eprintfn "%s" ex.Message
-                3
+                { RunResult.Default with ExitCode = ExitCode.IOError }
             | :? UnauthorizedAccessException as ex ->
                 eprintfn "%s" ex.Message
-                4
+                { RunResult.Default with ExitCode = ExitCode.UnauthorizedAccess }
 
-    and doMany arguments inputs =
-        inputs |> Seq.fold (fun (rtrnCode: int) filePath -> max rtrnCode (filePath |> doOne arguments)) 0
+    and foldResults (previousRunResult: RunResult) filePath =
```

Comment:
`doOneInput` and `doManyInputs` are `and`-ed together because they are mutually recursive, but `foldResults` can be independent. If you move its definition to above `doOneInput` then it doesn't need to be `and`-ed with the others. I'm not entirely sure, but I _think_ it is more efficient in F# to avoid `and`-ing when possible, but also it just disentangles otherwise independent pieces.